### PR TITLE
docs: add a getting started guide

### PR DESCRIPTION
**Pull request by gforsyth**
A draft that needs more work, but putting it up so we can iterate on it.

My thought was that we should have a sample DB to download and then connect to, as opposed to using `examples` (just for this, the examples are great)

### CODE REVIEW

**Comment by gforsyth** [comment]
Ok -- I finally got `gcloud` working and pushed up the palmer penguins sample database to the ibis examples bucket.
I think this is ready for a review.

File: docs/getting_started.md [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+Ibis can work with several file types, but at its core, it connects to existing
+databases and interacts with the data there. We'll use a local database
+(DuckDB) to get the hang of this.
+
+```python title="Download an example dataset"
+>>> import urllib.request
+>>> _ = urllib.request.urlretrieve(
+        "https://storage.googleapis.com/ibis-tutorial-data/palmer_penguins.ddb",
+        "palmer_penguins.ddb",
+         )
```
Comment (cpcloud) [inline docs/getting_started.md:35]: The indentation here looks a bit off, is it just me?

File: docs/index.md [hunk 2]
```diff
@@ -12,16 +12,16 @@ hide:
...
 [Tutorial](https://github.com/ibis-project/ibis-examples){ .md-button }
 </div>
 
 ---
 
 ```python title="Write high-level Python code"
 >>> import ibis
->>> con = ibis.connect('movielens.sqlite')
->>> movies = con.tables.movies
+>>> movies = ibis.examples.movielens.fetch()
```
Comment (cpcloud) [inline docs/index.md:24]:
```suggestion
>>> movies = ibis.examples.ml_latest_small_movies.fetch()
```

File: docs/getting_started.md [hunk 3]
```diff
@@ -0,0 +1,661 @@
...
+│ Dream     │             3900 │
+└───────────┴──────────────────┘
+```
+
+- What about the largest male penguin (by body mass) on each island for each year of data collection?
+
+```python
+>>> penguins.filter([penguins.sex == "male"]).group_by(["island", "year"]).aggregate(
+...:     penguins.body_mass_g.max().name("max_body_mass")
+...: ).order_by(["year", "max_body_mass"])
```
Comment (cpcloud) [inline docs/getting_started.md:641]: Many of these appear off by a single space. Can you remove the `:` after the `...`?

**Review by cpcloud — COMMENTED** [review]
Mostly style nits. I'm going to do a closer look using the rendered version.

File: docs/getting_started.md [hunk 4]
```diff
@@ -0,0 +1,661 @@
...
+
+## Download a (small) database file
+
+Ibis can work with several file types, but at its core, it connects to existing
+databases and interacts with the data there. We'll use a local database
+(DuckDB) to get the hang of this.
+
+```python title="Download an example dataset"
+>>> import urllib.request
+>>> _ = urllib.request.urlretrieve(
```
Comment (cpcloud) [inline docs/getting_started.md:32]: The underscore here looks a bit surprising, can we just let the thing return?

File: docs/getting_started.md [hunk 5]
```diff
@@ -0,0 +1,661 @@
...
+
+```shell title="Install Ibis using pip"
+$ pip install 'ibis-framework[duckdb]'
+```
+
+```shell title="Install Ibis using conda"
+$ conda install ibis-framework
+```
+
+## Download a (small) database file
```
Comment (cpcloud) [inline docs/getting_started.md:24]:
```suggestion
## Download a (small) database file
```

I don't think we need to qualify the size.

File: docs/getting_started.md (hunk 3 above)
Comment (gforsyth) [reply docs/getting_started.md]: Yep -- also removed all the `...:` -- I wanted these to be copy-pasteable so I overrode the `ipython` prompt, but forgot about the continuation lines

File: docs/index.md (hunk 2 above)
Comment (gforsyth) [reply docs/index.md]: done!

**Review by cpcloud — APPROVED** [review]
This is great, thanks for doing it!

🚢

File: docs/getting_started.md [hunk 6]
```diff
@@ -0,0 +1,661 @@
...
+Typing out ALL of the column names _except_ one is a little annoying. Instead of
+doing that again, we can use a `selector` to quickly select or deselect groups
+of columns.
+
+```python
+from ibis import selectors as s
+
+>>> penguins.mutate(bill_length_cm=penguins.bill_length_mm / 10).select(
+        ~s.matches("bill_length_mm")
+    )
```
Comment (jcrist) [inline docs/getting_started.md:336]: This could use a comment (or something) just to reiterate it's selecting all but `bill_length_mm`

**Review by jcrist — COMMENTED** [review]
Overall this looks good to me, just a few nits/questions.

**Comment by jcrist** [comment]
Gah, @kszucs merged while I was reviewing (no worries). I'm happy to push up a follow up PR with these changes, unless you want to handle them Gil.

**Comment by gforsyth** [comment]
> Gah, @kszucs merged while I was reviewing (no worries). I'm happy to push up a follow up PR with these changes, unless you want to handle them Gil.

I'll PR the updates

**Comment by jcrist** [comment]
Thanks!

**Pull request merged**